pkg/controller/worker/machines.go

```diff
@@ -140,13 +141,31 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 			})
 		}
 
+		resourceRequirements := &kubevirtv1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceCPU:    machineType.CPU,
+				v1.ResourceMemory: machineType.Memory,
+			},
+			OvercommitGuestOverhead: workerConfig.OvercommitGuestOverhead,
+		}
+
+		if w.cloudProfileConfig.MachineTypes != nil {
```

Comment:
Checking a slice for nil is not needed, a nil slice is an empty slice. Also, this case should be handled by `getMachineTypesExtension` so no need to do double work here.

Reply:
done